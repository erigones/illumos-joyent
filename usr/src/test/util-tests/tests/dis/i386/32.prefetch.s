/*
 * This file and its contents are supplied under the terms of the
 * Common Development and Distribution License ("CDDL"), version 1.0.
 * You may only use this file in accordance with the terms of version
 * 1.0 of the CDDL.
 *
 * A full copy of the text of the CDDL should have accompanied this
 * source.  A copy of the CDDL is also available via the Internet at
 * http://www.illumos.org/license/CDDL.
 */

/*
 * Copyright 2024 Oxide Computer Company
 */

/*
 * Test prefetch related instructions
 */

.text
.align 16
.globl libdis_test
.type libdis_test, @function
libdis_test:
	prefetch	(%eax)
	prefetcht0	(%esi)
	prefetcht1	(%ebx)
	prefetcht2	0x4(%ecx)
	prefetchw	0x8(%ecx)
	prefetchnta	0x167(%edx)
.size libdis_test, [.-libdis_test]
