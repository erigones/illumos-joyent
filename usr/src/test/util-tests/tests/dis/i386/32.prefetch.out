    libdis_test:      0f 0d 00           prefetch (%eax)
    libdis_test+0x3:  0f 18 0e           prefetcht0 (%esi)
    libdis_test+0x6:  0f 18 13           prefetcht1 (%ebx)
    libdis_test+0x9:  0f 18 59 04        prefetcht2 0x4(%ecx)
    libdis_test+0xd:  0f 0d 49 08        prefetchw 0x8(%ecx)
    libdis_test+0x11: 0f 18 82 67 01 00  prefetchnta 0x167(%edx)
                      00 
